ModTek/ProgressPanel.cs

```diff
@@ -56,8 +56,29 @@ private IEnumerator RunWorkList()
                 foreach (var workFunc in WorkList)
                 {
                     var workEnumerator = workFunc.Invoke();
-                    while (workEnumerator.MoveNext())
+                    bool didWork;
+
+                    do
                     {
+                        try
+                        {
+                            didWork = workEnumerator.MoveNext();
+                        }
+                        catch (Exception e)
+                        {
+                            Logger.FlushLogStream();
+                            Logger.Log($"");
+                            Logger.Log($"Uncaught ModTek exception!");
+                            Logger.Log(e.ToString());
+                            Logger.FlushLogStream();
```

Comment:
are there cases we don't want to flush? shouldnt we do it always? make a single log call, and then autoflush?